projects/plugins/jetpack/modules/masterbar/admin-menu/class-dashboard-switcher-tracking.php

```diff
@@ -0,0 +1,167 @@
+	 *
+	 * @var string
+	 */
+	private $plan;
+
+	/**
+	 * The wpcom_tracks wrapper function.
+	 *
+	 * @var callable
+	 */
+	private $wpcom_tracking;
+
+	/**
+	 * Dashboard_Switcher_Tracking constructor.
+	 *
+	 * @param Tracking $tracking       Jetpack tracking object.
+	 * @param callable $wpcom_tracking A wrapper over wpcom event record.
+	 * @param string   $plan           The current site plan.
+	 */
+	public function __construct( Tracking $tracking, callable $wpcom_tracking, $plan ) {
+		$this->tracking       = $tracking;
+		$this->plan           = $plan;
+		$this->wpcom_tracking = $wpcom_tracking;
+	}
+
+	/**
+	 * Create an event for the Quick switcher when the user changes it's preferred view.
+	 *
+	 * @param string $screen The screen page.
+	 * @param string $view   The new preferred view.
+	 */
+	public function record_switch_event( $screen, $view ) {
+		$event_props = array(
+			'current_page' => $screen,
+			'destination'  => $view,
+			'plan'         => $this->plan,
+		);
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$event_props['blog_id'] = get_current_blog_id();
+			$wpcom_tracking         = $this->wpcom_tracking;
+			$wpcom_tracking( $event_props );
+		} else {
+			$this->record_jetpack_event( $event_props );
+		}
+	}
+
+	/**
+	 * Get the current site plan or 'N/A' when we cannot determine site's plan.
+	 *
+	 * @todo: This method can be reused as a wrapper over WPCOM and Atomic as way to get site's current plan (display name).
+	 *
+	 * @return string
+	 */
+	public static function get_plan() {
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			if ( class_exists( '\WPCOM_Store_API' ) ) {
+				// @todo: Maybe introduce a wrapper for this since we are duplicating it from WPCOM_Admin_Menu:253
+				$products = \WPCOM_Store_API::get_current_plan( \get_current_blog_id() );
+				if ( array_key_exists( 'product_name_short', $products ) ) {
+					return $products['product_name_short'];
+				}
+			}
+
+			return 'N/A'; // maybe we should return free or null? At the moment it's safe to return 'N/A' since we use it only for passing it to the event.
+		}
+
+		// @todo: Maybe introduce a helper for this since we are duplicating it from Atomic_Admin_Menu:240
+		$products = Jetpack_Plan::get();
+		if ( array_key_exists( 'product_name_short', $products ) ) {
+			return $products['product_name_short'];
+		}
+
+		return 'N/A'; // maybe we should return free or null? At the moment we use it for passing it to the event.
+	}
+
+	/**
+	 * Record the event with Jetpack implementation.
+	 *
+	 * For Atomic sites we mark the Jetpack ToS option temporary as read.
+	 *
+	 * @todo Remove the jetpack_options_tos_agreed filter for Atomic sites after the Tracking is properly working for AT sites.
+	 *
+	 * @param array $event_properties The event properties.
+	 */
+	private function record_jetpack_event( $event_properties ) {
+		if ( \jetpack_is_atomic_site() ) {
+			add_filter( 'jetpack_options', array( __CLASS__, 'mark_jetpack_tos_as_read' ), 10, 2 );
```

Comment:
This is needed as a workaround for Atomic sites. I'll prepare a follow-up PR that proposes to fix the generic implementation we use in Jetpack.

Reply:
Don't understand why do we need this. Apparently [we don't check if the ToS are accepted as long as the user is connected to WP.com](https://github.com/Automattic/jetpack/blob/2b796f2e87c39bfd0b3ba7efb8d5859ae093a596/projects/packages/tracking/src/class-tracking.php#L224), which is always true on Atomic sites for the users who will see the quick switcher.